container/container.go

```diff
@@ -350,19 +350,20 @@ func (container *Container) GetExecIDs() []string {
 
 // Attach connects to the container's TTY, delegating to standard
 // streams or websockets depending on the configuration.
-func (container *Container) Attach(stdin io.ReadCloser, stdout io.Writer, stderr io.Writer, keys []byte) chan error {
+func (container *Container) Attach(stdin io.ReadCloser, stdout io.Writer, stderr io.Writer, keys []byte) (chan bool, chan error) {
 	ctx := container.InitAttachContext()
 	return AttachStreams(ctx, container.StreamConfig, container.Config.OpenStdin, container.Config.StdinOnce, container.Config.Tty, stdin, stdout, stderr, keys)
 }
 
 // AttachStreams connects streams to a TTY.
 // Used by exec too. Should this move somewhere else?
-func AttachStreams(ctx context.Context, streamConfig *runconfig.StreamConfig, openStdin, stdinOnce, tty bool, stdin io.ReadCloser, stdout io.Writer, stderr io.Writer, keys []byte) chan error {
+func AttachStreams(ctx context.Context, streamConfig *runconfig.StreamConfig, openStdin, stdinOnce, tty bool, stdin io.ReadCloser, stdout io.Writer, stderr io.Writer, keys []byte) (chan bool, chan error) {
 	var (
 		cStdout, cStderr io.ReadCloser
 		cStdin           io.WriteCloser
 		wg               sync.WaitGroup
 		errors           = make(chan error, 3)
+		detachChan       = make(chan bool, 1)
```

Comment:
+1, been holding off because I didn't like channel either, but didn't think of using an error to handle this.
I think that works.